File: backend/internal/domain/leaderboard.go
package domain

type LeaderboardEntry struct {
	Rank             int     `json:"rank"`
	UserID           int     `json:"user_id"`
	Username         string  `json:"username"`
	ProblemsSolved   int     `json:"problems_solved"`
	TotalSubmissions int     `json:"total_submissions"`
	AcceptanceRate   float64 `json:"acceptance_rate"`
}

File: frontend/src/features/users/api/leaderboard.ts
import { apiClient } from '@/shared/lib/axios'

export interface LeaderboardEntry {
    rank: number
    user_id: number
    username: string
    problems_solved: number
    total_submissions: number
    acceptance_rate: number
}

export const leaderboardApi = {
    getLeaderboard: (limit: number = 100) =>
        apiClient.get<LeaderboardEntry[]>(`/leaderboard?limit=${limit}`),
}
